demo/src/index.js

```diff
@@ -36,6 +39,51 @@ ${clauseText}
   return slateTransformer.fromMarkdown(defaultContractMarkdown);
 };
 
+/**
+ * Parses user inputted text for a template using Cicero
+ * @param {object} template The cicero template object.
+ * @param {string} text The user submitted text.
+ * @param {string} clauseId The uuid of the clause
+ * @returns {Promise} The result of the parse or an error.
+ */
+const parseClause = (template, text, clauseId) => {
+  try {
+    const ciceroClause = new Clause(template);
+    ciceroClause.parse(text);
+    const parseResult = ciceroClause.getData();
+    // Example code you could use to add this parse result to a redux store:
+    // store.dispatch(parseClauseSuccess(clauseId, parseResult));
+    return Promise.resolve(parseResult);
+  } catch (error) {
+    // Example code you could use to add this parse result to a redux store:
+    // store.dispatch(parseClauseError(clauseId, error));
+    return Promise.reject(error);
+  }
+};
+
+const fetchTemplateObj = async (action) => {
+  try {
+    const templateObj = await Template.fromUrl(action.uri);
```

Comment:
shouldn't the parameter just be templateUri and then pass that directly on line 66? when we use this method on line 109, we are just passing the uri, not an object.

Reply:
Correct, thank you for the catch!